src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPSelectionRangeSupport.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.selectionRange;
+
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LSPRequestConstants;
+import com.redhat.devtools.lsp4ij.LanguageServerItem;
+import com.redhat.devtools.lsp4ij.features.AbstractLSPDocumentFeatureSupport;
+import com.redhat.devtools.lsp4ij.internal.CancellationSupport;
+import com.redhat.devtools.lsp4ij.internal.CompletableFutures;
+import org.eclipse.lsp4j.SelectionRange;
+import org.eclipse.lsp4j.SelectionRangeParams;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * LSP selectionRange support which loads and caches selection ranges by consuming:
+ *
+ * <ul>
+ *     <li>LSP 'textDocument/selectionRange' requests</li>
+ * </ul>
+ */
+public class LSPSelectionRangeSupport extends AbstractLSPDocumentFeatureSupport<SelectionRangeParams, List<SelectionRange>> {
+
+    public LSPSelectionRangeSupport(@NotNull PsiFile file) {
+        super(file);
+    }
+
+    public CompletableFuture<List<SelectionRange>> getSelectionRanges(SelectionRangeParams params) {
+        return super.getFeatureData(params);
+    }
+
+    @Override
+    protected boolean isValidLSPFuture() {
+        // Disable caching until/unless it can take into account the params, in this case, the caret offset
+        return false;
```

Comment:
Is there another way to disable caching here? The issue is that the default caching is based on the file's modification count, but the response here is parameterized by the offset at the caret when the feature is invoked, so file-level caching causes issues. Is there a precedent for caching by both file modification count and some other parameter such as offset?